web-console/src/lib/services/pipelineManagerQuery.ts

```diff
@@ -83,8 +82,16 @@ export const PipelineManagerQuery = (({ pipelines, pipelineStatus, ...queries })
 }))(
   mkQuery({
     programs: () => ProgramsService.getPrograms(),
-    programCode: (programId: string) => ProgramsService.getProgram(programId, true),
-    programStatus: (programId: string) => ProgramsService.getProgram(programId, false).then(v => v),
+    // programCode: (programName: string) => ProgramsService.getProgram(programName, true), // TODO: Uncomment with an API update
+    programCode: (programName: string) =>
+      ProgramsService.getPrograms(undefined, programName, true).then(p =>
+        p.length ? p[0] : Promise.reject(new Error('/v0/programs: ' + programName + ' not found'))
```

Comment:
Why the /v0? Who consumes this error?

Reply:
In a previous implementation when programCode with a given programId was not found (should be super rare) an error was not handled, so it would bubble up (and shown in a bottom left red error popup, i think). Here I am manually throwing an error to keep this behavior.

Reply:
The error message is not 1 to 1 to previous one, just did a simple reasonable one

Reply:
But why the "/v0/programs" is part of the error?

Reply:
This "v0" is an internal thing that no one needs to know about. I don't think it should be part of the error message

Reply:
`/v0/programs` is the REST API endpoint. I agree with Mihai that we probably don't need to share that in the error message string.

Reply:
@lalithsuresh will api return 4XX if /v0/programs?name=... doesn't find a program? I assumed it would return empty list.

Reply:
Upd: It does return 4xx. Patched the custom error out